server/test/services/export/JsonExporterTest.java

```diff
@@ -29,61 +36,70 @@ public void testAllQuestionTypesWithoutEnumerators() throws Exception {
                 SubmittedApplicationFilter.EMPTY)
             .getLeft();
     ResultAsserter resultAsserter = new ResultAsserter(resultJsonString);
-    System.out.println(resultJsonString);
 
     resultAsserter.assertLengthOf(3);
     testApplicationTopLevelAnswers(fakeProgram, resultAsserter, applicationOne, 2);
     resultAsserter.assertValueAtPath("$[2].status", STATUS_VALUE);
-    resultAsserter.assertValueAtPath(2, ".applicant_name.first_name", "Alice");
-    resultAsserter.assertNullValueAtPath(2, ".applicant_name.middle_name");
-    resultAsserter.assertValueAtPath(2, ".applicant_name.last_name", "Appleton");
-    resultAsserter.assertValueAtPath(2, ".applicant_birth_date.date", "1980-01-01");
-    resultAsserter.assertValueAtPath(2, ".applicant_email_address.email", "one@example.com");
-    resultAsserter.assertValueAtPath(2, ".applicant_address.zip", "54321");
-    resultAsserter.assertValueAtPath(2, ".applicant_address.city", "city");
-    resultAsserter.assertValueAtPath(2, ".applicant_address.street", "street st");
-    resultAsserter.assertValueAtPath(2, ".applicant_address.state", "AB");
-    resultAsserter.assertValueAtPath(2, ".applicant_address.line2", "apt 100");
-    resultAsserter.assertValueAtPath(1, ".applicant_phone.phone_number", "+16157571010");
-    resultAsserter.assertValueAtPath(
+    resultAsserter.assertValueAtApplicationPath(2, ".applicant_name.first_name", "Alice");
```

Comment:
I feel like this test would be significantly easier to read if it were asserting the state of one json document, instead of asserting each line, like:

https://github.com/civiform/civiform/blob/f39bc2d0b984cfdd3280b458f8a548fc6f770513/server/test/services/export/QuestionJsonSamplerTest.java#L156-L172

But understood that's probably out of scope here.

Reply:
Agreed, hence #5257, which should remove this large test altogether. Currently, after that change the tests will still test the document via the resultAsserter methods instead of comparing the raw json directly, but comparing raw json definitely has benefits. (I haven't so far because I only want to test the specific question in a given test, not the whole json document, and I didn't want to write a method to extract the document below a given path.)

The only change here is renaming `resultAsserter.assertValueAtPath()` to `resultAsserter.assertValueAtApplicationPath()`. Sorry, I thought the diff would highlight what changed _within_ the line to make that more clear.

Reply:
SG. Yeah, GitHub should definitely have character diffing by now...